Fixes 'yarn start' on Windows - LavaPack is not defined (#13318)

HowardBraham opened the pull request:
Fixes #13318, Fixes #13793, Fixes #14984, and Fixes #15237
Discussed with @kumavis and @danjm
Please let me know if this breaks another system and a more complicated fix is necessary.

Update 11/18/2022: Also adds a getPathInsideNodeModules helper function to end reliance on writing paths that start with `./node_modules/`

HowardBraham commented:
I have read the CLA Document and I hereby sign the CLA

legobeat reviewed (dismissed):
Can not speak for Windows. Does not break anything on Linux.

matthewwalsh0 commented:
Not certain if it's the same issue, but I hit a similar error building on Windows previously and resolved it by updating the copy paths before they reached `fast-glob`. 

```
async function copyGlob(baseDir, srcGlob, dest) {
    const fixedSrcGlob =
      process.platform === 'win32' ? srcGlob.replace(/\\/gu, '/') : srcGlob;

    const sources = await glob(fixedSrcGlob, { onlyFiles: false });

    await Promise.all(
      sources.map(async (src) => {
        const relativePath = path.relative(baseDir, src);
        await fs.copySync(src, `${dest}${relativePath}`);
      }),
    );
  }
 ```

HowardBraham committed:
Fixes 'yarn start' on Windows - LavaPack is not defined (#13318)

Also adds a getPathInsideNodeModules helper function to end reliance on writing paths that start with './node_modules/'

legobeat commented on development/build/utils.js:
```diff
@@ -117,10 +118,38 @@ function logError(error) {
+    targetPath += '/';
+  }
+
+  targetPath += pathToFiles;
```
Is there a reason to not do `path.join` here? Would also remove the need for manually inserting `/` above.

legobeat commented on development/build/utils.js:
```diff
@@ -117,10 +118,38 @@ function logError(error) {
+  targetPath += pathToFiles;
+
+  if (process.platform === 'win32') {
+    targetPath = targetPath.replace(/\\/gu, '/');
```
Would [`path.normalize`](https://nodejs.org/api/path.html#pathnormalizepath) be helpful here? 

Also [`path.sep`](https://nodejs.org/api/path.html#pathsep)

HowardBraham replied:
`path.normalize`, or even `path.posix.normalize` will not work for this.  Here's a discussion about it: https://github.com/nodejs/node/issues/12298

I replaced with the following and pushed a new commit
`targetPath.split(path.sep).join(path.posix.sep)`

HowardBraham committed:
Implementing legobeat's suggestions, modified slightly

legobeat commented on development/build/static.js:
```diff
@@ -127,11 +128,14 @@ function getCopyTargets(shouldIncludeLockdown) {
     },
     {
-      src: `./node_modules/react-responsive-carousel/lib/styles`,
+      src: getPathInsideNodeModules('react-responsive-carousel', 'lib/styles'),
```
above we do `images/` - any reason not to be consistent with trailing `/`?

HowardBraham replied:
Yeah good call

legobeat reviewed (approved):
One minor thing, LGTM otherwise!

legobeat commented on development/build/static.js:
```diff
@@ -127,11 +128,14 @@ function getCopyTargets(shouldIncludeLockdown) {
     {
-      src: `./node_modules/@fortawesome/fontawesome-free/webfonts/`,
+      src: getPathInsideNodeModules(
+        '@fortawesome/fontawesome-free',
```
```suggestion
        '@fontawesome/fontawesome-free',
```
typo?

legobeat replied:
Nope, nvm!

HowardBraham committed:
added a trailing slash

legobeat committed:
Merge branch 'develop' into fix/13318-lavapack-is-not-defined

jpuri commented:
I tested the PR on mac and `yarn start` works well 👍

Gudahtt reviewed (approved):
LGTM!
